### PR TITLE
Fix column option readonly

### PR DESCRIPTION
**Pull request by hielkehoeve**
As described in issue #2041 and issue #2221 the readonly option is not working correctly.

I have fixed and updated de testcase to use `.save` and `.update` and fixed the code accordingly.

### CODE REVIEW

**Comment by hielkehoeve** [comment]
Is this framework still alive?

**Comment by hielkehoeve** [comment]
...

**Comment by nightwolfz** [comment]
Looks like no one is around :(

**Comment by nightwolfz** [comment]
Looks like no one is around :(

**Comment by pleerock** [comment]
Sorry for a long response here.

We need to think how it should work now.

1. when we call `save` if it does insert it should only insert the entity, if it does update it should skip this entity column value

2. we have UpdateQueryBuilder that can execute any update query. Should we get in count readonly in there? From one side we can update an entity using update query builder, from another side update query builder should work like regular SQL and let you update anything.

Maybe for second we can disallow readonly columns updation, but add some method into UpdateQueryBuilder like `readOnlyColumns(boolean)` ?

adding @Kononnable

File: src/persistence/Subject.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export class Subject {
             if (this.metadata.isJunction && changeMap.column) {
                 valueMap = changeMap.column.createValueMap(changeMap.column.referencedColumn!.getEntityValue(value));
 
-            } else if (changeMap.column) {
+            } else if (changeMap.column && (!changeMap.column.isReadonly || !this.databaseEntity)) {
```
[the remark this answers is not in the document]
Comment (hielkehoeve) [reply src/persistence/Subject.ts]: if: the column is not readonly OR the action is an insert action.

File: test/functional/columns/readonly/columns-readonly.ts [hunk 2]
```diff
@@ -34,9 +34,20 @@ describe("columns > readonly functionality", () => {
...
         expect(loadedPost!.text).to.be.equal("Some text about columns1");
-        expect(loadedPost!.authorName).to.be.equal("Umed1");
+        expect(loadedPost!.authorName).to.be.equal("Umed");
 
-    })));
 
+        // then update all its properties and save again
+        post.title = "About columns1";
+        post.text = "Some text about columns1";
+        post.authorName = "Umed1";
```
Comment (Kononnable) [inline test/functional/columns/readonly/columns-readonly.ts]: I think this should throw an error. If property is readonly we should get an error on assignment. Just ignoring such situations may lead to bugs in app logic.

Comment (ejaustin) [reply test/functional/columns/readonly/columns-readonly.ts]: Any updates here? Reading `tests` code and saw this bug.
`expect(loadedPost!.authorName).to.be.equal("Umed1");`

I'd expect `post.authorName` NOT be updated and get an error or at least a consistent expected behavior.

**Comment by imnotjames** [comment]
There's conflicts and tests now show this feature as working in master, going to close.

**Pull request closed without merging**